src/ott/geometry/costs.py

```diff
@@ -280,6 +280,45 @@ def twist_operator(
       return vec + jax.grad(self.h_legendre)(-dual_vec)
     return vec - jax.grad(self.h_legendre)(dual_vec)
 
+  def transport_map(self,
+                    f: Func) -> Callable[[jnp.ndarray, bool, Any], jnp.ndarray]:
+    r"""Get an optimal transport map for a concave function :math:`f`.
+
+    Uses Theorem 1.17 from :cite:`santambrogio:15` to define an OT map, e.g. in
+    the forward case :math:`x - (\nabla h^*) \circ \nabla \bar f^h(x)`, where
+    :math:`h^*` is the Legendre transform of :math:`h` and :math:`\bar f^h`
+    is the :meth:`h_transform` of a concave function :math:`f`.
+
+    Args:
+      f: Concave function.
+
+    Returns:
+      The transport map with a signature ``(x, forward, **kwargs)``.
+    """
+
+    def transport(
+        x: jnp.ndarray, forward: bool = True, **kwargs: Any
+    ) -> jnp.ndarray:
+      """Transport points from source to the target or vice-versa.
+
+      Args:
+        x: Array of shape ``[n, d]``.
+        forward: Whether to transport the points from source to the target
```

Comment:
i am wondering if this works.

intuitively, when one has access to a pair of approximate optimal dual potentials, say `f*` and `g*`, then one can go forward or backward by choosing either potential. Here, it seems that you are using the same potential twice. So in principle this can be used to transport a source to a target, but I don't think you can claim the ``forward=False`` will be approximately the inverse of ``forward``. So in a sense, I think there can be only one direction?

Reply:
Agree, you're completely right, removed the `forward` option.